plugins/dashboard/__init__.py

```
@@ -745,16 +750,34 @@ def edit_plot(self, item):
         self._data[item.name] = data
         self.apply_data()
 
-    def load_plot_data(self, plot_id):
-        item = self.get_item(plot_id)
-        if item is None:
-            return {}
+    def load_plot_data(self, item_name, refetch=False):
+        item = self.get_item(item_name)
+        if not refetch:
+            cached_data = self.load_cached_data(item.to_cache_key())
+            if cached_data:
+                return cached_data
 
         data = self.load_plot_data_for_item(item)
-        if isinstance(data, dict):
-            return data
+        self.cache_data(item.to_cache_key(), data)
 
-        return {}
+        return data
+
+    def load_cached_data(self, run_key):
+        try:
+            results = self.view.load_run_results(run_key)
+            if results:
+                return results.get("data", None)
+        except:
+            return None
+        return None
+
+    def cache_data(self, item_name, data):
+        try:
+            run_key = item_name
+            results = self.ctx.dataset.init_run_results(run_key, data=data)
+            self.ctx.dataset.save_run_results(run_key, results)
+        except:
```

This should probably not ignore all errors... but what error should it ignore?